server/src/categories/categories.controller.ts

```diff
@@ -0,0 +1,95 @@
+import {
+    Controller,
+    Get,
+    Post,
+    Body,
+    Patch,
+    Param,
+    Delete,
+    UseGuards,
+    HttpException,
+    HttpStatus,
+  } from '@nestjs/common';
+  
+  import { CategoriesService } from './categories.service';
+  import { CreateCategoryDto } from './dto/create-category';
+  import { UpdateCategoryDto } from './dto/update-category';
+  // import { JwtAuthGuard } from '../auth/guards/jwt-auth.guard';
+  //authGaurd excluded because it is outside of the scope of this PR
+  import { DeleteResult } from 'typeorm';
+  import { Category } from './entities/category.entity';
+  
+  @Controller('categories')
+  export class CategoriesController {
+    constructor(private readonly CategoriesService: CategoriesService) {}
+  
+    @Post()
+    async create(
+      @Body() createCategoryDto: CreateCategoryDto,
+    ): Promise<Category | HttpException> {
+      try {
+        const newCategory = await this.CategoriesService.create(createCategoryDto);
+        return newCategory;
+      } catch (error) {
+        throw new HttpException(
+          { status: HttpStatus.CONFLICT, message: `${error}` },
+          HttpStatus.CONFLICT,
+        );
+      }
+    }
+  
+    // @UseGuards(JwtAuthGuard)
+    @Get()
+    findAll(): Promise<Category[]> {
+      return this.CategoriesService.findAll();
+    }

```

Comment:
similar to in your service how you have `async func():...` but don't have `await`s in there, we should have them here
this function is still async but the `return` is acting as our `await`, it's implied since we cannot return until we get the response back from the CategoriesService
same with the other functions in this file